File: q3397/q3397.java
class Solution {
    public int maxDistinctElements(int[] nums, int k) {
        Arrays.sort(nums);
        int distinct = 0;
        int prevMin = Integer.MIN_VALUE; 
        for (int num : nums) {
            int currMin = Math.min(Math.max(num - k, prevMin + 1), num + k);
            if (currMin > prevMin) {
                distinct++;
                prevMin = currMin;
            }
        }
        return distinct;
    }
}

File: q3397/q3397.py
class Solution:
    def maxDistinctElements(self, nums: List[int], k: int) -> int:
        nums.sort()
        prevMin = -math.inf
        distinct = 0
        for num in nums:
            currMin = min(max(num - k, prevMin + 1), num + k)
            if currMin > prevMin: 
                distinct += 1
                prevMin = currMin
        return distinct
